opennebula/resource_opennebula_virtual_machine.go

```diff
@@ -1504,6 +1535,27 @@ func generateVm(d *schema.ResourceData, tplContext *dyn.Vector) (string, error)
 }
 
 func resourceVMCustomizeDiff(diff *schema.ResourceDiff, v interface{}) error {
+
+	onChange := diff.Get("on_disk_change").(string)
+	oldDisk, newDisk := diff.GetChange("disk")
+
+	oldDiskList := oldDisk.([]interface{})
+	newDiskList := newDisk.([]interface{})
+
+	for i, oi := range oldDiskList {
+		oiB := oi.(map[string]interface{})
+		niB := newDiskList[i].(map[string]interface{})
```

Comment:
If the user remove a disk newDiskList will be shorter of 1 element, and `i` will point outside of the slice. To do this you'll have to check the slice length